### ISSUE TITLE
Compact Map issue

### ISSUE THREAD

**Issue by arjungupta0107** [RocketChat/Rocket.Chat.iOS#1616]
Native iOS App:
- Your Rocket.Chat app version: 2.3.0
- Device (or Simulator) you're running with: Simulator( iPhone 6s 11.2)

I am unable to run the code. After installation of pod, i am trying to run the code on simulator, but getting 5 errors regarding compact map in share extension.

File Wise errors and respected screenshots are given below:
 
 1.  ../Rocket.Chat.iOS/Rocket.Chat.ShareExtension/State/ActionCreators/SelectServer.swift:12:42: Value of type '[[String : String]]' has no member 'compactMap'

![error1](https://user-images.githubusercontent.com/37107527/39424722-9cf27836-4c95-11e8-974c-ed9a50774314.png)

2.  ../Rocket.Chat.iOS/Rocket.Chat/Models/Message/MessageReaction.swift:21:32: Value of type '[JSON]' has no member 'compactMap'

![error2](https://user-images.githubusercontent.com/37107527/39424469-7e8c35cc-4c94-11e8-8f6a-3157f0c1d8ad.png)

3. ../Rocket.Chat.iOS/Rocket.Chat.ShareExtension/Base/SENavigationController.swift:14:98: Value of type '[Any]' has no member 'compactMap' 

![error3](https://user-images.githubusercontent.com/37107527/39424499-a9e2ce5c-4c94-11e8-931b-8a8a88545e00.png)

4.  ../Rocket.Chat.iOS/Rocket.Chat.ShareExtension/State/ActionCreators/SubmitContent.swift:52:30: Value of type '[(Int, SEStore.Request)]' has no member 'compactMap'

![error4](https://user-images.githubusercontent.com/37107527/39424537-d3e1aec6-4c94-11e8-960b-a9976717a760.png)

5.  ../Rocket.Chat.iOS/Rocket.Chat.ShareExtension/State/ActionCreators/SubmitContent.swift:102:33: Value of type '[(Int, SEStore.Request)]' has no member 'compactMap'

![error5](https://user-images.githubusercontent.com/37107527/39424578-f5387a28-4c94-11e8-8437-9f76b97e9c40.png)

**Comment by artrmz** [comment]
Are you building it using XCode 9.3?

**Comment by arjungupta0107** [comment]
No, i am using XCode 9.2.

**Comment by Sameesunkaria** [comment]
@arjungupta0107 `compactMap` had been introduced with swift 4.1 (Xcode 9.3), to replace and deprecate `flatMap`. The functionality of both is the same, but the `compactMap` name better describes its operation. 

TL;DR
**Make sure you have Xcode 9.3 or up**

**Comment by rafaelks** [comment]
Thanks for the help @Sameesunkaria and @artrmz! Closing the issue for now.
